src/s3/util/fileViewerManager.ts

```diff
@@ -78,86 +92,135 @@ export class S3FileViewerManager {
                     return
                 }
                 await this.removeAndCloseTab(activeTab)
-                await this.openTab(fileNode)
+                await this.openInReadMode(fileNode)
                 this._onDidChange.fire(activeTab.s3Uri)
             }
         })
+    }
 
-        this.window.onDidChangeVisibleTextEditors(editors => {
-            const editorSet = new Set(editors)
-            for (const value of this.activeTabs.values()) {
-                //if visible text editors don't contain a given S3Tab anymore,
-                //set the S3Tab.editor to undefined
-                if (value.editor) {
-                    if (!editorSet.has(value.editor)) {
-                        value.editor = undefined
-                    }
-                }
-            }
-        })
+    public async focusAndCloseTab(
+        uri: vscode.Uri,
+        editor?: vscode.TextEditor,
+        workspace = vscode.workspace
+    ): Promise<void> {
+        if (!editor) {
+            const doc = await workspace.openTextDocument(uri)
+            await this.window.showTextDocument(doc, {
+                preview: false,
+            })
+        } else {
+            await this.window.showTextDocument(editor.document, {
+                preview: false,
+                viewColumn: editor.viewColumn,
+            })
+        }
```

Comment:
This can probably be shortened to 
```
            const doc = editor ? editor.document : (await workspace.openTextDocument(uri))
            await this.window.showTextDocument(doc, {
                preview: false,
                viewColumn: editor?.viewColumn,
            })